### DISCUSSION TITLE
static lifetime for label?

### DISCUSSION THREAD

**Discussion by dimitrove** [fltk-rs/fltk-rs#811] [Q&A] [answered]
I am very new in Rust and can not figure out why first function compiles and works and the second does not compile and requires static lifetime for the parameter...

fn title_pack(title: &str) -> group::Pack {
    let mut hpack = group::Pack::default()
        .with_type(group::PackType::Horizontal).with_size(0, 24);
    hpack.end();
    hpack.add(
        &( frame::Frame::default()
            .with_pos(0, 0)
            .with_size(1, 0)
            .with_label(title)
            .with_align(enums::Align::RightBottom)
        )
    );
    hpack
}

// xxx
fn title_pack2(title: &str) -> group::Pack {
    let mut hpack = group::Pack::default()
        .with_type(group::PackType::Horizontal).with_size(0, 24);
    hpack.end();
    hpack.add(
        &( frame::Frame::new(0, 0, 1, 0, title)
            .with_align(enums::Align::RightBottom)
        )
    );
    hpack
}

**Comment by MoAlyousef** [comment, marked as answer]
FLTK’s widget constructor expects a static lifetime, so this is reflected in fltk-rs post version 1.0. Previously we used to just leak the string, now we just require the ::new constructor requires a static lifetime, while the with_label and set_label don’t require that. Thus the first example works and the second doesn’t. You can change the signature of the second function to take a `title: &'static str` and it should work.

**Reply by dimitrove** [reply]
Thank you very much. I will check what "leak the string" means, but do with_label and set_label "leak the string"? Will they be changed to become like the ::new constructor?

**Reply by MoAlyousef** [reply]
No problem. 
No set_label and with_label don't leak, and they won't change to become like the ::new constructor.
